source/allThen.js

```diff
@@ -0,0 +1,51 @@
+import _curry2 from './internal/_curry2.js';
+import _isString from './internal/_isString.js';
+import _isArrayLike from './internal/_isArrayLike.js';
+import _isFunction from './internal/_isFunction.js';
+import _of from './internal/_of.js';
+import andThen from './andThen.js';
+
+var symbolIterator = Symbol == null || Symbol.iterator == null ? 'Symbol(Symbol.iterator)' : Symbol.iterator;
+
+var _isIterator = function(a) {
+  return a[symbolIterator] != null;
+};
+
+/**
+ * Returns a single Promise that resolves to an array of the results of the input promises.
+ *
+ * @func
+ * @memberOf R
+ * @since v0.27.1
+ * @category Function
+ * @sig ([a] -> b) -> a -> (Promise e b)
+ * @sig ([a] -> b) -> [a] -> (Promise e b)
+ * @sig ([a] -> b) -> [(Promise e a)] -> (Promise e b)
+ * @param {Function} f The function which will be called with an array.
+ * @param {*} ps a value or an array.
+ * @return {Promise} The result of calling `Promise.all(ps).then(f)`
+ * @example
+ *
+ *      const p1 = s => Promise.resolve(s + 1);
+ *      const p2 = s => Promise.resolve(s + 2);
+ *
+ *      const  = R.pipe(
+ *        R.of,
+ *        R.ap([p1, p2]),
+ *        R.allThen(R.join('|'))  // => 'test1|test2'
+ *      );
```

Comment:
Although we're not entirely consistent with this, I think it's better if our examples rarely use any Ramda functions outside the one being demonstrated.  While `join` is probably clear to most readers, both `of` and `ap` might cause confusion.

Perhaps better would be something like this:

```js
R .allThen (xs => xs .join ('|')) ([Promise .resolve ('foo'), Promise .resolve ('bar')])
  .then (console .log) //~> "foo|bar"
```

Reply:
Thanks, it's really more concise.